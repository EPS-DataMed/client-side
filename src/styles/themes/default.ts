export const defaultTheme = {
  colors: {
    neutral: '#FFFFFF',
    neutral50: '#F0F0F0',
    neutral100: '#F9F9F9',
    neutral200: '#E5E5E5',
    neutral300: '#CCCCCC',
    neutral400: '#A5A5A5',
    neutral500: '#7F7F7F',
    neutral600: '#42424A',
    neutral700: '#36363B',
    neutral900: '#181818',
    blue500: '#2980B9',
    blue440: '#34CDFF',
    green500: '#3fd255',
    red500: '#FF0000',
  },
  fontSizes: {
    xxs: '0.625rem',
    xs: '0.75rem',
    sm: '0.875rem',
    md: '1rem',
    lg: '1.125rem',
    xl: '1.25rem',
    '2xl': '1.5rem',
    '4xl': '2rem',
    '5xl': '2.25rem',
    '6xl': '3rem',
    '7xl': '4rem',
    '8xl': '4.5rem',
    '9xl': '6rem',
  },
  fontWeights: {
    regular: '400',
    medium: '500',
    bold: '700',
  },
  fonts: {
    primary: 'Open Sans, sans-serif',
    secondary: 'Poppins, sans-serif',
  },
  lineHeights: {
    shorter: '125%',
    short: '140%',
    base: '160%',
    tall: '180%',
  },
  radii: {
    px: '1px',
    xs: '4px',
    sm: '6px',
    md: '8px',
    lg: '16px',
    full: '99999px',
  },
  space: {
    1: '0.25rem',
    2: '0.5rem',
    3: '0.75rem',
    4: '1rem',
    5: '1.25rem',
    6: '1.5rem',
    7: '1.75rem',
    8: '2rem',
    9: '2.25rem',
    10: '2.5rem',
    12: '3rem',
    16: '4rem',
    19: '4.75rem',
    20: '5rem',
    27: '6.75rem',
    40: '10rem',
    64: '16rem',
    80: '20rem',
  },
} as const
