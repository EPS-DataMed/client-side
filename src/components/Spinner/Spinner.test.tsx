import { render } from '@testing-library/react'
import 'jest-styled-components'
import { ThemeProvider } from 'styled-components'
import { Spinner } from './' // Ajuste o caminho conforme necessÃ¡rio
import { defaultTheme } from '../../styles/themes/default'

describe('Spinner Component', () => {
  it('should render the spinner component', () => {
    const { container } = render(
      <ThemeProvider theme={defaultTheme}>
        <Spinner />
      </ThemeProvider>,
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
