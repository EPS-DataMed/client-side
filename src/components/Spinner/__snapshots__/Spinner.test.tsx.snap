// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner Component should render the spinner component 1`] = `
.c0 {
  animation: fvtopB 1s linear infinite;
  width: 12px;
  height: 12px;
}

<svg
  class="c0"
  fill="none"
  viewBox="0 0 16 16"
  xmlns="http://www.w3.org/2000/svg"
>
  <defs>
    <filter
      id="grayFilter"
    >
      <fecolormatrix
        type="matrix"
        values="0.33 0.33 0.33 0 0
                    0.33 0.33 0.33 0 0
                    0.33 0.33 0.33 0 0
                    0    0    0    0.5 0"
      />
    </filter>
  </defs>
  <path
    d="M16 8C16 12.4183 12.4183 16 8 16C3.58172 16 0 12.4183 0 8C0 3.58172 3.58172 0 8 0C12.4183 0 16 3.58172 16 8ZM2.4 8C2.4 11.0928 4.90721 13.6 8 13.6C11.0928 13.6 13.6 11.0928 13.6 8C13.6 4.90721 11.0928 2.4 8 2.4C4.90721 2.4 2.4 4.90721 2.4 8Z"
    fill="white"
    opacity="0.1"
  />
  <path
    d="M14.8 8C15.4627 8 16.0092 7.45979 15.9102 6.80449C15.8136 6.16523 15.6396 5.53856 15.391 4.93853C14.989 3.96793 14.3997 3.08601 13.6569 2.34315C12.914 1.60028 12.0321 1.011 11.0615 0.608964C10.4614 0.360423 9.83477 0.186448 9.19551 0.0898325C8.54021 -0.0092077 8 0.537258 8 1.2C8 1.86274 8.54326 2.38715 9.19084 2.52808C9.51571 2.59878 9.8344 2.69844 10.143 2.82627C10.8225 3.1077 11.4398 3.52019 11.9598 4.0402C12.4798 4.56021 12.8923 5.17755 13.1737 5.85697C13.3016 6.1656 13.4012 6.48429 13.4719 6.80916C13.6129 7.45674 14.1373 8 14.8 8Z"
    fill="white"
  />
</svg>
`;
